db/indexes_test.go

```diff
@@ -57,53 +59,42 @@ func TestInitializeIndexes(t *testing.T) {
 
 			// Make sure we can drop and reinitialize twice
 			for i := 0; i < 2; i++ {
-				dropErr := base.DropAllIndexes(base.TestCtx(t), n1qlStore)
-				require.NoError(t, dropErr, "Error dropping all indexes")
-
-				initErr := InitializeIndexes(n1qlStore, test.xattrs, 0, true)
-				require.NoError(t, initErr, "Error initializing all indexes")
-
-				// Recreate the primary index required by the test bucket pooling framework
-				err := n1qlStore.CreatePrimaryIndex(base.PrimaryIndexName, nil)
-				require.NoError(t, err)
-
-				validateErr := validateAllIndexesOnline(b)
-				require.NoError(t, validateErr, "Error validating indexes online")
+				err := dropAndInitializeIndexes(base.TestCtx(t), n1qlStore, b, test.xattrs)
+				require.NoError(t, err, "Error dropping and initialising all indexes on bucket")
 			}
 		})
 	}
 
 }
 
 // Reset bucket state
-func validateAllIndexesOnline(bucket base.Bucket) error {
+func validateAllIndexesOnline(bucket base.Bucket, xattrs bool) error {
 
-	n1QLStore, ok := base.AsN1QLStore(bucket)
-	if !ok {
-		return fmt.Errorf("Bucket is not gocb bucket: %T", bucket)
-	}
+	col, err := base.AsCollection(bucket)
 
-	// Retrieve all indexes
-	getIndexesStatement := fmt.Sprintf("SELECT indexes.name, indexes.state from system:indexes where keyspace_id = %q", n1QLStore.GetName())
-	results, err := n1QLStore.Query(getIndexesStatement, nil, base.RequestPlus, true)
-	if err != nil {
-		return err
-	}
+	cluster := col.GetCluster()
+	mgr := cluster.QueryIndexes()
 
-	var indexRow struct {
-		Name  string
-		State string
+	watchOption := gocb.WatchQueryIndexOptions{
+		WatchPrimary:   true,
+		ScopeName:      "sg_test_1",
+		CollectionName: "sg_test_1",
 	}
 
-	for results.Next(&indexRow) {
-		if indexRow.State != base.IndexStateOnline {
-			return fmt.Errorf("Index %s is not online", indexRow.Name)
-		} else {
-			log.Printf("Validated index %s is %s", indexRow.Name, indexRow.State)
+	// Watch and wait some time for indexes to come online
+	if xattrs {
+		err = mgr.WatchIndexes(bucket.GetName(), []string{"#primary", "sg_access_x1", "sg_allDocs_x1", "sg_channels_x1", "sg_roleAccess_x1", "sg_syncDocs_x1", "sg_tombstones_x1"}, 10*time.Second, &watchOption)
```

Comment:
Construct these like:


https://cs.github.com/couchbase/sync_gateway/blob/7e9bc42ecbf2198af01bda918448265772e231f6/db/indexes.go?q=path%3Aindexes.go#L292

just create a function to return all indexes names since it is used twice